crud.py

```diff
@@ -0,0 +1,164 @@
+from sqlalchemy.orm import Session
+from sqlalchemy import text
+from fastapi import HTTPException, Response, status
+
+import models, schemas, utils
+
+
+def get_author(db: Session, author_id: int):
+    return db.query(models.DBAuthor).filter(models.DBAuthor.id == author_id).first()
+
+
+def get_author_list(
+    db: Session,
+    skip: int = 0,
+    limit: int = 10,
+    author_name: str | None = None,
+    sort_field: str | None = None,
+):
+    queryset = db.query(models.DBAuthor)
+    if author_name is not None:
+        queryset = queryset.filter(models.DBAuthor.name.contains(author_name))
+    if sort_field is not None:
+        if sort_field.startswith("-"):
+            sort_field = sort_field[1:] + " desc"
+        queryset = queryset.order_by(text(sort_field))
+    return queryset.offset(skip).limit(limit).all()
+
+
+def create_author(
+    db: Session,
+    author: schemas.AuthorCreate
+):
+    new_author = models.DBAuthor(
+        name=author.name,
+        bio=author.bio,
+    )
+    db.add(new_author)
+    db.commit()
+    db.refresh(new_author)
+    return new_author
+
+
+def get_author_by_name(db: Session, authorname: str):
+    return db.query(models.DBAuthor).filter(
+        models.DBAuthor.name == authorname
+        ).first()
+
+
+def get_book(db: Session, book_id: int):
+    return db.query(models.DBBook).filter(
+        models.DBBook.id == book_id
+        ).first()
+
+
+def get_book_list(
+    db: Session,
+    skip: int = 0,
+    limit: int = 10,
+    author_id: int | None = None,
+    book_title: str | None = None,
+    sort_field: str | None = None
+):
+    queryset = db.query(models.DBBook)
+    if author_id is not None:
+        queryset = queryset.filter(models.DBBook.author_id == author_id)
+    if book_title is not None:
+        queryset = queryset.filter(models.DBBook.title.contains(book_title))
+    if sort_field is not None:
+        if sort_field.startswith("-"):
+            sort_field = sort_field[1:] + " desc"
+        queryset = queryset.order_by(text(sort_field))
+    return queryset.offset(skip).limit(limit).all()
+
+
+def create_book(db: Session, book: schemas.BookCreate):
+    new_book = models.DBBook(**book.dict())
+    db.add(new_book)
+    db.commit()
+    db.refresh(new_book)
+    return new_book
+
+
+def get_book_by_title(db: Session, title: str):
+    return db.query(models.DBBook).filter(models.DBBook.title == title).first()
+
+
+def patch_book(
+    book_id: int,
+    db: Session,
+    book: schemas.BookCreate
+):
+    db_book = get_book(db, book_id=book_id)
+    if db_book is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Book was not found"
+        )
+    book_data = book.dict(exclude_unset=True)
+    for key, val in book_data.items():
+        setattr(db_book, key, val)
+    db.add(db_book)
+    db.commit()
+    db.refresh(db_book)
+    return db_book
```

Comment:
many of your functions in this module have way too much responsibilities. They are checking for existence, handling errors, and perform db actions. Don't forget about SRP :)